Sources/SwiftProtobuf/SwiftProtobufContiguousBytes.swift

```diff
@@ -0,0 +1,21 @@
+// Sources/SwiftProtobuf/SwiftProtobufContiguousBytes.swift
+//
+// Copyright (c) 2022 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/main/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+
+public protocol SwiftProtobufContiguousBytes {
```

Comment:
Can we add some docs to this stating why we need it and what owners of other bag of bytes type should do with it?